packages/data-context/src/sources/migration/legacyOptions.ts

```diff
@@ -0,0 +1,305 @@
+interface ResolvedConfigOption {
+  name: string
+  defaultValue?: any
+  isFolder?: boolean
+  isExperimental?: boolean
+  /**
+   * Can be mutated with Cypress.config() or test-specific configuration overrides
+   */
+  canUpdateDuringTestTime?: boolean
+}
+
+interface RuntimeConfigOption {
+  name: string
+  defaultValue: any
+  isInternal?: boolean
+  /**
+   * Can be mutated with Cypress.config() or test-specific configuration overrides
+   */
+  canUpdateDuringTestTime?: boolean
+}
+
+// NOTE:
+// If you add/remove/change a config value, make sure to update the following
+// - cli/types/index.d.ts (including allowed config options on TestOptions)
+// - cypress.schema.json
+//
+// Add options in alphabetical order for better readability
+
+// TODO - add boolean attribute to indicate read-only / static vs mutable options
```

Comment:
Do we have a followup ticket for this?

Reply:
no we don't. this file is a pure copy paste of the 9.X `config/lib/options.ts`. I should clean it up and remove all the not so useful fluff from it.

I could also add where it comes from at the top of the file with a warning to use it only to migrate 9.X projects.